src/tests/URITests.cpp

```diff
@@ -0,0 +1,86 @@
+#include "URI.h"
+#include <gtest/gtest.h>
+
+using namespace uri;
+
+TEST(URI, host) {
```

Comment:
Better test names for some of the tests.
E.g. host_only_gives_host_plus_default_port

There are others below too